test/NVMLAcceleratorTopoTest.cpp

```diff
@@ -0,0 +1,398 @@
+    }
+    EXPECT_CALL(*m_device_pool, num_accelerator()).WillOnce(Return(num_accelerator));
+
+    NVMLAcceleratorTopo topo(*m_device_pool, num_cpu);
+
+    EXPECT_EQ(4, topo.num_accelerator());
+    std::set<int> cpus_allowed_set[topo.num_accelerator()];
+    cpus_allowed_set[0] = {0,1,2,3,4,5,6,7,8,9};
+    cpus_allowed_set[1] = {10,11,12,13,14,15,16,17,18,19};
+    cpus_allowed_set[2] = {20,21,22,23,24,25,26,27,28,29};
+    cpus_allowed_set[3] = {30,31,32,33,34,35,36,37,38,39};
+
+    for (int accel_idx = 0; accel_idx < topo.num_accelerator(); ++accel_idx) {
+        ASSERT_THAT(topo.ideal_cpu_affinitization(accel_idx), cpus_allowed_set[accel_idx]);
+    }
+}
+
+//Test case:  Accel N+1 associates with all CPUs of Accel N, but not vice versa
+TEST_F(NVMLAcceleratorTopoTest, n1_superset_n_affinitization_config)
+{
+    int num_accelerator = 4;
+    int num_cpu = 40;
+
+    unsigned long accel_bitmask[num_accelerator];
+    accel_bitmask[0] = 0xfffffff000;
+    accel_bitmask[1] = 0xffffffff00;
+    accel_bitmask[2] = 0xfffffffff0;
+    accel_bitmask[3] = 0xffffffffff;
+
+    for (int accel_idx = 0; accel_idx < num_accelerator; ++accel_idx) {
+        EXPECT_CALL(*m_device_pool, ideal_cpu_affinitization_mask(accel_idx)).WillOnce(Return((cpu_set_t *)&accel_bitmask[accel_idx]));
+    }
+    EXPECT_CALL(*m_device_pool, num_accelerator()).WillOnce(Return(num_accelerator));
+
+    NVMLAcceleratorTopo topo(*m_device_pool, num_cpu);
+
+    EXPECT_EQ(4, topo.num_accelerator());
+    std::set<int> cpus_allowed_set[topo.num_accelerator()];
+    cpus_allowed_set[0] = {12,13,14,15,16,17,18,19,20,21};
+    cpus_allowed_set[1] = {8 ,9 ,10,11,22,23,24,25,26,27};
+    cpus_allowed_set[2] = {4 ,5 ,6 ,7 ,28,29,30,31,32,33};
+    cpus_allowed_set[3] = {0 ,1 ,2 ,3 ,34,35,36,37,38,39};
+
+    for (int accel_idx = 0; accel_idx < topo.num_accelerator(); ++accel_idx) {
+        ASSERT_THAT(topo.ideal_cpu_affinitization(accel_idx), cpus_allowed_set[accel_idx]);
+    }
+}
+
+//Test case:  Last accelerator has the smallest map, and the entire map will be 'stolen' to cause starvation
+TEST_F(NVMLAcceleratorTopoTest, greedbuster_affinitization_config)
+{
+    int num_accelerator = 4;
+    int num_cpu = 40;
+
+    unsigned long accel_bitmask[num_accelerator];
+    accel_bitmask[0] = 0xffffffffff;
+    accel_bitmask[1] = 0xfffffffff0;
+    accel_bitmask[2] = 0x0fffffff00;
+    accel_bitmask[3] = 0x00000003ff;
+
+    for (int accel_idx = 0; accel_idx < num_accelerator; ++accel_idx) {
+        EXPECT_CALL(*m_device_pool, ideal_cpu_affinitization_mask(accel_idx)).WillOnce(Return((cpu_set_t *)&accel_bitmask[accel_idx]));
+    }
+    EXPECT_CALL(*m_device_pool, num_accelerator()).WillOnce(Return(num_accelerator));
+
+    EXPECT_THROW(NVMLAcceleratorTopo topo(*m_device_pool, num_cpu), geopm::Exception);
+
+    //TODO: If we move away from the greedy affinitization approach this is the expected result.
```

Comment:
commented code.  Just test for current behavior.  you can also open an issue to change this behavior later.